cluster-state-service/canary/pkg/tests/get_instance.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/blox/blox/cluster-state-service/canary/pkg/tests/util"
+	"github.com/blox/blox/cluster-state-service/canary/pkg/wrappers"
+	"github.com/blox/blox/cluster-state-service/swagger/v1/generated/client/operations"
+	"github.com/pkg/errors"
+)
+
+const (
+	getInstanceMetric            = "getInstance"
+	getInstanceClusterNamePrefix = canaryClusterNamePrefix + "GetInstance_"
+)
+
+func (c canary) GetInstance() error {
+	clusterName := getInstanceClusterNamePrefix + util.TimeString()
+
+	err := c.ecsWrapper.CreateCluster(&clusterName)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Get instance - created cluster")
+
+	numInstances := int64(1)
+	ec2InstanceIDs, err := c.ec2Wrapper.LaunchInstances(&numInstances, &clusterName)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Get instance - launched EC2 instance")
+
+	instanceARN, err := util.ValidateECSContainerInstanceAndGetInstanceARN(c.ecsWrapper, clusterName)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Get instance - instance registered with ECS")
+
+	err = validateGetInstanceCall(c.cssWrapper, clusterName, instanceARN)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Get instance - instance in CSS GetInstance response")
+
+	err = util.DeleteCluster(c.ecsWrapper, clusterName)
+	if err != nil {
+		return err
+	}
+
+	err = util.TerminateInstances(c.ec2Wrapper, ec2InstanceIDs)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Get instance - GetInstance test resources cleaned up")
+
+	return nil
+}
+
+func (c canary) GetInstanceMetric() string {
+	return getInstanceMetric
+}
+
+func validateGetInstanceCall(cssWrapper wrappers.CSSWrapper, clusterName string, instanceARN string) error {
+	// Takes some time for CSS to get the instance.
+	// Retry get call once every 10 seconds for 2 minutes.
+	found := false
+	for i := 0; i < 12; i++ {
```

Comment:
add a todo to move to constants if you want to do it in a later commit (here and in all other files)